core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessage.java

```diff
@@ -299,6 +306,9 @@ void notifySubscriber(SubscriptionImpl subscription, CompletableFuture<?> comple
                     if (subscription.notifyCancellation || !(cause instanceof CancelledSubscriptionException)) {
                         subscriber.onError(cause);
                     }
+                } catch (Exception e) {
+                    logger.warn("Subscriber.onError() should not raise an exception. subscriber: {}",
```

Comment:
We can probably create an exception that wraps `cause` and complete the future with it for this case so the exception isn't lost.

Reply:
Ah that's a good point. Let me fix this.

Reply:
So you mean we have to define a new sort of comp0site exception which has the `cause` and `e` here, in this case?

Reply:
sort of [this one](http://reactivex.io/RxJava/javadoc/io/reactivex/exceptions/CompositeException.html)?

Reply:
Yeah I was thinking something like that :)

Reply:
Just copied the code from them because it seems they implemented well. 😄

Reply:
Can't we just use `Throwable.addSuppressed()`?

Reply:
What should we do if we catch a fatal exception? Perhaps add the less fatal one using `addSuppressed()` and throw?

Reply:
Never mind. It looks OK.